backend/dal/Migrations/v01.08.01/Up/PreUp/01-Buildings.sql

```diff
@@ -0,0 +1,423 @@
+    AND p.[AgencyId] = @CMBId
+
+UPDATE p
+    SET p.[AgencyId] = (SELECT [Id] FROM dbo.[Agencies] WHERE [Code] = 'SD 10')
+FROM dbo.[Buildings] p
+JOIN dbo.[Addresses] a ON p.AddressId = a.Id
+WHERE a.[AdministrativeArea] IN ('Arrow Lakes', 'Edgewood', 'Burton', 'Shoreholme', 'Silverton', 'New Denver', 'Nakusp', 'Fauquier')
+    AND p.[AgencyId] = @CMBId
+
+UPDATE p
+    SET p.[AgencyId] = (SELECT [Id] FROM dbo.[Agencies] WHERE [Code] = 'SD 19')
+FROM dbo.[Buildings] p
+JOIN dbo.[Addresses] a ON p.AddressId = a.Id
+WHERE a.[AdministrativeArea] IN ('Revelstoke', 'Beaton', 'Mica Creek')
+    AND p.[AgencyId] = @CMBId
+
+UPDATE p
+    SET p.[AgencyId] = (SELECT [Id] FROM dbo.[Agencies] WHERE [Code] = 'SD 20')
+FROM dbo.[Buildings] p
+JOIN dbo.[Addresses] a ON p.AddressId = a.Id
+WHERE a.[AdministrativeArea] IN ('Castlegar', 'Rossland', 'Trail', 'Montrose', 'Warfield', 'Fruitvale', 'Renata', 'Robson', 'Glade')
+    AND p.[AgencyId] = @CMBId
+
+UPDATE p
+    SET p.[AgencyId] = (SELECT [Id] FROM dbo.[Agencies] WHERE [Code] = 'SD 22')
+FROM dbo.[Buildings] p
+JOIN dbo.[Addresses] a ON p.AddressId = a.Id
+WHERE a.[AdministrativeArea] IN ('Vernon', 'Lumby', 'Coldstream', 'Mabel Lake', 'Cherryville', 'Oyama')
+    AND p.[AgencyId] = @CMBId
+
+UPDATE p
+    SET p.[AgencyId] = (SELECT [Id] FROM dbo.[Agencies] WHERE [Code] = 'SD 23')
+FROM dbo.[Buildings] p
+JOIN dbo.[Addresses] a ON p.AddressId = a.Id
+WHERE a.[AdministrativeArea] IN ('Central Okanagan', 'Kelowna', 'Peachland', 'Lake Country', 'Westbank', 'West Kelowna')
+    AND p.[AgencyId] = @CMBId
+
+UPDATE p
+    SET p.[AgencyId] = (SELECT [Id] FROM dbo.[Agencies] WHERE [Code] = 'SD 27')
+FROM dbo.[Buildings] p
+JOIN dbo.[Addresses] a ON p.AddressId = a.Id
+WHERE a.[AdministrativeArea] IN ('150 Mile House', 'Riske Creek', 'Bridge Lake', 'Mcleese Lake', 'Likely', 'Ulkatcho', 'Anahim Lake', 'Towdystan', 'Kleena Kleene', 'Tatla Lake', 'Redstone', 'Alexis Creek', 'Williams Lake', '100 Mile House', 'Horsefly', 'Quesnel Lake', 'Canim Lake', 'Mahood Lake', 'Bridge Lake')
+    AND p.[AgencyId] = @CMBId
+
+UPDATE p
+    SET p.[AgencyId] = (SELECT [Id] FROM dbo.[Agencies] WHERE [Code] = 'SD 28')
+FROM dbo.[Buildings] p
+JOIN dbo.[Addresses] a ON p.AddressId = a.Id
+WHERE a.[AdministrativeArea] IN ('Quesnel', 'Nazko', 'Wells')
+    AND p.[AgencyId] = @CMBId
+
+UPDATE p
+    SET p.[AgencyId] = (SELECT [Id] FROM dbo.[Agencies] WHERE [Code] = 'SD 33')
+FROM dbo.[Buildings] p
+JOIN dbo.[Addresses] a ON p.AddressId = a.Id
+WHERE a.[AdministrativeArea] IN ('Chilliwack', 'Abbotsford', 'Rosedale', 'Cultus Lake', 'Sardis', 'Deroche', 'Errock')
+    AND p.[AgencyId] = @CMBId
+
+UPDATE p
+    SET p.[AgencyId] = (SELECT [Id] FROM dbo.[Agencies] WHERE [Code] = 'SD 34')
+FROM dbo.[Buildings] p
+JOIN dbo.[Addresses] a ON p.AddressId = a.Id
+WHERE a.[AdministrativeArea] IN ('Abbotsford')
```

Comment:
SD 33 also has Abbotsford in the Where, wouldn't this mean no buildings would be put in SD 34?

Reply:
Nice catch